bitbot/reminder.go

```diff
@@ -0,0 +1,259 @@
+package bitbot
+
+import (
+	"fmt"
+	"github.com/whyrusleeping/hellabot"
+	log "gopkg.in/inconshreveable/log15.v2"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+//nolint gochecknoglobals
+var (
+	location   *time.Location
+	timeFormat string
+)
+
+type eventType struct {
+	ID          int `gorm:"unique;AUTO_INCREMENT;PRIMARY_KEY"`
+	Channel     string
+	Author      string
+	Description string
+	People      string
+	Time        time.Time
+}
+
+var ReminderTrigger = NamedTrigger{ //nolint:gochecknoglobals,golint
+	ID:   "reminder",
+	Help: "Set up events and remind them to concerned people. Usage: !remind list|time|add|remove|join|part",
+	Condition: func(irc *hbot.Bot, m *hbot.Message) bool {
+		return m.Command == "PRIVMSG" && strings.HasPrefix(m.Trailing, "!remind")
+	},
+	Action: func(irc *hbot.Bot, m *hbot.Message) bool {
+		timeFormat = "2006-01-02 15:04"
+
+		var err error
+		location, err = time.LoadLocation("UTC")
+		if err != nil {
+			irc.Reply(m, "Something went wrong: Couldn't load timezone")
+			log.Error("Reminder : Couldn't load UTC timezone", err.Error())
+		}
+
+		b.DB.AutoMigrate(&eventType{})
```

Comment:
Automigrate should only happen when the bot starts, not whenever the function is called. We should setup an init funtion for the DB that gets called when the bot initializes.

Reply:
I'm not satisfied with my current solution (see `autoMigrateDB` in `bitbot/bot.go`, but I'm out of ideas.